### PR TITLE
TrayPublisher: colorspace overrides from defined ocio config

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Adding enumerator attribute which is having aggregated options from active ocio config.

# Testing notes
## test global config
1. activate `project_settings/global/imageio/activate_global_color_management`
2. make sure some value is set here with correct path to ocio path `project_settings/global/imageio/ocio_config/filepath/0`
3. test and see values are aggregated form the config

## test host config override
1. activate `project_settings/global/imageio/activate_global_color_management`
2. activate `project_settings/traypublisher/imageio/activate_host_color_management` 
3. activate `project_settings/traypublisher/imageio/ocio_config/override_global_config`
4. make sure different config path is defined here ideally with different colorspace names `project_settings/traypublisher/imageio/ocio_config/filepath/0`
5. Reload Publisher
6. test and see values are aggregated form the config

## test deactivated colorspace management with default values
1. deactivate `project_settings/global/imageio/activate_global_color_management`
2. Reload Publisher
3. See default values in enumerator with 4 generic options.

### CODE REVIEW

File: openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py [hunk 1]
```diff
@@ -27,10 +33,38 @@ def process(self, instance):
...
+                (name, name)
+                for name, family in config_items.items()
+            ))
+        else:
+            cls.colorspace_items.extend([
+                ("sRGB", "sRGB"),
+                ("rec709", "rec709"),
+                ("ACES", "ACES"),
+                ("ACEScg", "ACEScg")
+            ])
```
Comment (BigRoy) [inline openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py]: Are we sure this makes sense? Isn't it weird to default to some random entries like this? I know these are popular entries, but they might not exist at all in certain ocio configs. Would it make more sense to maybe just disable the EnumDef entry completely (like hide it?) and log a debug message if no OCIO config is set with an explanation as to why the feature is disabled?

File: openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py [hunk 2]
```diff
@@ -27,10 +33,38 @@ def process(self, instance):
...
+            project_settings=project_settings
+        )
+
+        if config_data:
+            filepath = config_data["path"]
+            config_items = colorspace.get_ocio_config_colorspaces(filepath)
+            cls.colorspace_items.extend((
+                (name, name)
+                for name, family in config_items.items()
+            ))
```
Comment (BigRoy) [inline openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py]:
```suggestion
            cls.colorspace_items.extend((
                (name, name) for name in config_items.keys()
            ))
```

File: openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py [hunk 3]
```diff
@@ -13,9 +14,14 @@ class CollectColorspace(pyblish.api.InstancePlugin,
...
+    colorspace_items = [
+        (None, "Don't override")
+    ]
+
     def process(self, instance):
         values = self.get_attr_values_from_data(instance.data)
         colorspace = values.get("colorspace", None)
+        self.log.debug("colorspace: {}".format(colorspace))
         if not colorspace:
             return
```
Comment (BigRoy) [inline openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py]: Should we log this only after `if not colorspace: return` below so we don't log "Don't override"?


```suggestion
        if colorspace is None:
            return

        self.log.debug("Explicit colorspace set to: {}".format(colorspace))
```

**Review by BigRoy — COMMENTED** [review]
Nice job, just some small comments!

How is the OCIO colorspaces retrieval speed?

File: openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py (hunk 1 above)
Comment (jakubjezek001) [reply openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py]: good point, make sense to just hide it.

**Review by BigRoy — APPROVED** [review]
Thanks, solid work. Code looks great!

**Pull request merged**